gateway/api/views.py

```diff
@@ -114,16 +116,63 @@ def get_model_serializer_run_program(*args, **kwargs):
     def get_serializer_class(self):
         return self.serializer_class
 
+    def get_object(self):
+        logger.warning("ProgramViewSet.get_object not implemented")
+
     def get_queryset(self):
-        # Allow unauthenticated users to read the swagger documentation
-        if self.request.user is None or not self.request.user.is_authenticated:
-            return Program.objects.none()
-        return (
-            Program.objects.all().filter(author=self.request.user).order_by("-created")
+        author = self.request.user
+
+        logger.info("ProgramViewSet get view_program permission")
+        view_program_permission = Permission.objects.get(
+            codename=VIEW_PROGRAM_PERMISSION
         )
 
-    def perform_create(self, serializer):
-        serializer.save(author=self.request.user)
+        # Groups logic
+        user_criteria = Q(user=author)
+        author_criteria = Q(author=author)
```

Comment:
just doublechecking that user and author criteria are the same

Reply:
They are not exactly exactly the same in the sense that:

```
Program:
  author
  
Group:
  user
```

They are different fields in database so you need different criterias. If we had `user` instead of `author` we could reuse it.

Reply:
it looks like @Tansito is:
- first, getting list of groups that user belongs to and program as permissions from
- then he selects programs that has this groups or user is the author of the program here https://github.com/Qiskit-Extensions/quantum-serverless/pull/1306/files#diff-2025150bf47484f2582f9816af400b1336b7ca6adcbb4faa61bb2d3330df856dR151

Am I right, @Tansito ?

Reply:
Exactly, maybe reordering the blocks and put author criteria in programs logic is clearer than now.

Reply:
Hope this change helps to clarify it:

[6ce8d69](https://github.com/Qiskit-Extensions/quantum-serverless/pull/1306/commits/6ce8d69d39d6d8b95f714585253f123173bdcb65)